### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by mntedev**
### **We have completed the following tasks, :hand:** 

:heavy_check_mark:  Resolved  n + 1 problem for a post with the author
:heavy_check_mark:  Resolved n + 1 problem for a post with comments and author
:heavy_check_mark:  Added  integration specification for user listing feature
:heavy_check_mark:  Added integration specification for user detail feature
:heavy_check_mark:  Added integration specification to user posts listing feature
:heavy_check_mark:  Added integration specification to user post detail feature

### CODE REVIEW

File: spec/features/post_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      expect(page).to have_content @posts.first.author.name
+    end
+    it 'Should show the number fo comments given for the post' do
+      expect(page).to have_content 'Comments: 3'
+    end
+    it 'Should show the numbers of like for post' do
+      expect(@posts.first.likes_counter).to be 0
+    end
+    it 'Should show the post text' do
+      expect(page).to have_content @posts.first.text
```
Comment (uwadonat) [inline spec/features/post_show_spec.rb:28]: - [ ] Some tests are failing. please fix these:
Screenshot from 2022-09-27 22-32-03

**Review by uwadonat — CHANGES_REQUESTED** [review]
## Hi @Team,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- Descriptive pull request message ✔️
- No linter errors ✔️
almoo

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by mntedev** [comment]
1. I changed the name to resolve ambiguous 
2.  Run the test and the result look like 

image

**Review by ShoiraTa — APPROVED** [review]
## **STATUS: APPROVED  🟢 ** 
------
### Hi @worashf , 

**Great work on making the changes requested by a previous reviewer 👏🏻** 

### Highlights:
 ✔️ Project is setup correctly
 ✔️ All linters are passing
 ✔️ Correct gitflow
 ✔️ Professional documentation

Your project is complete! . There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

image

**Cheers and Happy coding! :clap: :clap: :clap:**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@shoirata] in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
***
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have  limited number of  reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**